src/Decompiler/Evaluation/UnaryNegEqZeroRule.cs

```diff
@@ -0,0 +1,56 @@
+#region License
+/*
+ * Copyright (C) 2020 Sven Almgren.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+
+using Reko.Core;
+using Reko.Core.Expressions;
+using Reko.Core.Operators;
+using Reko.Core.Types;
+using System;
+
+namespace Reko.Evaluation
+{
+    public class UnaryNegEqZeroRule
```

Comment:
Please add a clarifying comment here. We should also be quite explicit about the code assuming that two's complement representation is being used here. Who knows what -0 means on one's complement machines.